blockly/src/server/Server.java

```diff
@@ -1364,32 +1368,36 @@ public PositionComponent getHeroPosition() {
    *
    * <p>The hero needs at least one unit of ammunition to successfully shoot a fireball.
    */
-  public void shootFireBall() {
-    AmmunitionComponent heroAC =
-        hero.fetch(AmmunitionComponent.class)
-            .orElseThrow(() -> MissingComponentException.build(hero, AmmunitionComponent.class));
-
-    if (heroAC.checkAmmunition()) {
-      utils.Direction viewDirection =
-          convertPosCompDirectionToUtilsDirection(EntityUtils.getViewDirection(hero));
-      Skill fireball =
-          new Skill(
-              new FireballSkill(
-                  () -> {
-                    CollideComponent collider =
-                        hero.fetch(CollideComponent.class)
-                            .orElseThrow(
-                                () ->
-                                    MissingComponentException.build(hero, CollideComponent.class));
-
-                    Point start = collider.center(hero);
-                    return start.add(new Point(viewDirection.x(), viewDirection.y()));
-                  }),
-              1);
-      fireball.execute(hero);
-      heroAC.spendAmmo();
-      waitDelta();
-    }
+  public void shootFireball() {
+    hero.fetch(AmmunitionComponent.class).stream()
+        .filter(AmmunitionComponent::checkAmmunition)
+        .forEach(ac -> aimAndShoot(ac));
+  }
+
+  /**
+   * Shoots a fireball in direction the hero is facing.
+   *
+   * @param ac AmmunitionComponent of the hero, ammunition amount will be reduced by 1
+   */
+  private void aimAndShoot(AmmunitionComponent ac) {
+    if (!ac.checkAmmunition()) return;
+    utils.Direction viewDirection =
+        convertPosCompDirectionToUtilsDirection(EntityUtils.getViewDirection(hero));
+    Skill fireball =
+        new Skill(
+            new FireballSkill(
+                () ->
+                    hero.fetch(CollideComponent.class)
+                        .map(cc -> cc.center(hero).add(viewDirection.toPoint()))
```

Comment:
wenn du das in zwei operationen auseinander ziehst, dann wird das lesbarer.

Reply:
versteh ich nicht

Reply:
``` java
.map(cc -> cc.center(hero))
.map(p -> p.add(viewDirection.toPoint()))
```

Reply:
i c